Brand-new tooltipish Helpbox for ProjectBrowser

brusherru opened the pull request:
It closes #930.

evgenykochetkov commented on packages/xod-client/src/core/styles/abstracts/variables.scss:
```diff
@@ -6,8 +6,8 @@ $font-family-display: 'Roboto Slab', sans-serif;
-$font-size-l: 18px;
-$font-size-xl: 28px;
+$font-size-l: 14px;
+$font-size-xl: 18px;
```
👍🏻

nkrkv commented:
:heart: Few issues:

- [ ] The scrollbar in a help tooltip is native and ugly. Can we change it to the XOD one?
- [ ] When a project browser item goes out of the visible area I expect the tooltip to close automatically (equivalent to (x) click). It does not happen.

![help-bug](https://user-images.githubusercontent.com/146383/34202542-12adcfea-e589-11e7-9d3a-8e2cbb2175f1.gif)

nkrkv commented on packages/xod-client/src/core/selectors.js:
```diff
@@ -29,56 +29,25 @@ export const hasUnsavedChanges = createSelector(
-      maybeSelectedPathInProjectBrowser,
-    ]);
-  }
+export const getPatchforHelpbox = createSelector(
```
Capitalize “For”?

nkrkv commented on packages/xod-client/src/core/selectors.js:
```diff
@@ -29,56 +29,25 @@ export const hasUnsavedChanges = createSelector(
+    Maybe
+  )(selectedPatchPath)
+);
+export const getPatchforQuickHelp = createSelector(
```
Capitalize “For”?

evgenykochetkov commented on packages/xod-client/src/projectBrowser/components/PatchGroupItemContextMenu.jsx:
```diff
@@ -5,7 +5,14 @@ import { ContextMenu, MenuItem, connectMenu } from 'react-contextmenu';
 import { PATCH_GROUP_CONTEXT_MENU_ID } from '../constants';
 
-const callCallbackWithPatchPath = onClick => (event, data) => onClick(data.patchPath);
+const callCallbackWithPatchPath = onClick => (event, data) => {
```
I think we should rename this function 🤔

nkrkv commented on packages/xod-client/src/core/styles/components/Helpbox.scss:
```diff
@@ -0,0 +1,59 @@
+
+  &-content {
+    padding: 12px;
+    overflow: auto; // TODO
```
TODO what? Does this thing cause native scrollbars?

brusherru replied:
Yep, forget it 😬

nkrkv commented on packages/xod-client/src/editor/constants.js:
```diff
@@ -95,3 +95,7 @@ export const SIDEBAR_IDS = {
+
+// Event name for pub/sub to update position of Helpbox
+// while User scrolls ProjectBrowser or selects another Patch
+export const UPDATE_HELPBOX_POSITION = 'UPDATE_HELPBOX_POSITION';
```
Should we better call it `SCROLL_PROJECT_BROWSER`? That’s the fact, and updating the helpbox position is one of the reactions (although the only one at the moment)

brusherru replied:
In fact, it listens to scroll and click events, so it's not just "SCROLL"...
But you're right, that's only about updating position from ProjectBrowser.

I propose to leave it as is and rename it to a more appropriate name later (during #932) when it will receive update events from the quick search.

brusherru commented:
All fixed, except renaming of `UPDATE_HELPBOX_POSITION`.

Also did small fixes of sidebar system (when any Panel was opened on a whole sidebar it's not fits window height and the latest item was out of the viewport, so I use flexbox and problem disappeared) and rename `Dock To Right` into `Dock to Right` and `to Dock Left`. :)

nkrkv commented:
I have circle background behind (x) disappeared. Can you bring it back?

Also, I’m not sure we should expose the disappeared helpbox again when the scrolled item is visible again. It is questionable. Seems like nothing bad about the current behavior, but imagine a scenario when a user invokes help, scroll the item out so that it disappears and does other things for five minutes. Then it scrolls the project browser and boom :boom: helpbox appears. He already forgot about it. Wouldn’t it look like a bug?

brusherru commented:
> Then it scrolls the project browser and boom 💥 helpbox appears. He already forgot about it.

Sounds reasonable. Fixed it.
And fixed background behind cross in close button.
